lib/Monad_WP/TraceMonad.thy

```diff
@@ -941,54 +941,17 @@ where
   "\<lbrace>P\<rbrace> f -,\<lbrace>Q\<rbrace> \<equiv> validE P f (\<lambda>x y. True) Q"
 
 
-text \<open> Abbreviations for trivial preconditions: \<close>
-abbreviation(input)
-  top :: "'a \<Rightarrow> bool" ("\<top>")
-where
-  "\<top> \<equiv> \<lambda>_. True"
-
-abbreviation(input)
-  bottom :: "'a \<Rightarrow> bool" ("\<bottom>")
-where
-  "\<bottom> \<equiv> \<lambda>_. False"
-
-text \<open> Abbreviations for trivial postconditions (taking two arguments): \<close>
-abbreviation(input)
-  toptop :: "'a \<Rightarrow> 'b \<Rightarrow> bool" ("\<top>\<top>")
-where
-  "\<top>\<top> \<equiv> \<lambda>_ _. True"
-
+text \<open> Abbreviations for trivial postconditions (taking three arguments): \<close>
 abbreviation(input)
   toptoptop :: "'a \<Rightarrow> 'b \<Rightarrow> 'b \<Rightarrow> bool" ("\<top>\<top>\<top>")
 where
   "\<top>\<top>\<top> \<equiv> \<lambda>_ _ _. True"
```

Comment:
silly consistency question: why can we get rid of toptop, but not toptoptop?

Reply:
All monad Hoare logics introduced toptop, but only the TraceMonad uses toptoptop, and I didn't want to introduce something into the wider context that we should instead get rid of. 

I haven't figured out yet how to get rid of toptop (and therefore toptoptop) without too much impact on how wp works, that's why it still exists at all. But removing it should be the next step in this particular quest.